en/sensor/sf45_rotating_lidar.md

```diff
@@ -0,0 +1,125 @@
+| <a id="SF45_ORIENT_CFG"></a>[SF45_ORIENT_CFG](../advanced_config/parameter_reference.md#SF45_ORIENT_CFG)    | Set the orientation of the sensor (facing up or down)    |
+| <a id="SF45_UPDATE_CFG"></a>[SF45_UPDATE_CFG](../advanced_config/parameter_reference.md#SF45_UPDATE_CFG)    | Set the update rate                                      |
+| <a id="SF45_YAW_CFG"></a>[SF45_YAW_CFG](../advanced_config/parameter_reference.md#SF45_YAW_CFG)             | Set the yaw orientation                                  |
+
+## Testing
+
+You can confirm that the sensor is correctly configured by connecting QGroundControl, and observing that [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE) is present in the [MAVLink Inspector](https://docs.qgroundcontrol.com/master/en/qgc-user-guide/analyze_view/mavlink_inspector.html).
+
+The obstacle overlay in QGC will look like this:
+
+![sf45 obstacle avoidance map shown in QGC](../../assets/hardware/sensors/lidar_lightware/sf45_obstacle_map.png)
+
+## Driver Implementation
+
+The [sensor driver](../modules/modules_driver_distance_sensor.md#lightware-sf45-serial) publishes the [ObstacleDistance](../msg_docs/ObstacleDistance.md) UORB Message that is used by PX4 [Collision Prevention](../computer_vision/collision_prevention.md).
+The measurements in each sector will correspond to the lowest measurement the sensor had in that corresponding sector.
+The data is then published to the [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE) MAVLink message.
+
+## Troubleshooting
+
+### Common problems
+
+Start-up issues, jerky movements, or a lot of communication errors displayed with `lightware_sf45_serial status` could mean that the sensor is not getting enough power.
+
+According to its datasheet, the sensor needs 300 mA of current at 5V.
+The supplied cable is fairly long and when connected to the 5V supply, such as the flight controller `TELEM2` port, the voltage at the rangefinder may drop below the required level.
+One mitigation/alternative is to power the SF45 via a separate step-down converter from battery voltage, ensuring that there are 5V across the rangefinder itself.
+
+### Debugging with PlotJuggler
+
+Debugging problems with distance sensors is much easier if you can plot the [OBSTACLE_DISTANCE](https://mavlink.io/en/messages/common.html#OBSTACLE_DISTANCE) message in an x-y plot, because this lets you directly see the orientation of the measurement in respect of the drone.
+
+The video below shows how you can view such a plot in PlotJuggler.
+
+<video src="../../assets/hardware/sensors/lidar_lightware/sf45_plotjuggler.mp4" width="720" controls></video>
+
+In order to generate this kind of plot you will need to add the following Reactive Scripts to PlotJuggler.
+If you save these scripts and then add new data, you will see a new timeseries called `obstacle_distance_xy`, which is the same as is displayed in the video.
+
+**Global code, executed once:**
```

Comment:
Is it bleedingly obvious how you execute/add these scripts? If not then we need to either add a link to instructions or explain how here. Up to you which.

Reply:
Sadly not, i'll add a screenshot, and mention where the setting is.

Reply:
@Claudio-Chies I'm going to merge this, but ...

I've added a tip above in https://github.com/PX4/PX4-user_guide/pull/3448/commits/55fcd83cc06ea56d49dd78606ce539b67cb26a51 to link to the more extensive example you provided in Collision prevention.

Do we need both sets of scripts and examples? In the tip I have said "go here for more info" since that shows how to get scripts working, but they do more or less the same thing right? I wonder if having two scripts like this makes sense? [why wouldn't you just use the collision prevention stuff?]